pre-commit

```diff
@@ -0,0 +1,82 @@
+#!/bin/sh
+# original source: https://raw.githubusercontent.com/streadway/amqp/master/pre-commit
+
+GOFMT_OK=1
+GOLINT_OK=1
+GOVET_OK=1
+
+main() {
+    mv_vendor_out
+    run_gofmt
+    run_golint
+    mv_vendor_in
+    run_govet
+    check_results
+}
+
+local_go_version_is_latest_stable() {
+  go version | grep -q $LATEST_STABLE_SUPPORTED_GO_VERSION
+}
+
+log_error() {
+  echo "$*" 1>&2
+}
+
+mv_vendor_out() {
```

Comment:
These functions exist because we don't want the testing (like `gofmt`) to fail because our vendored items don't pass muster. Unfortunately you can't just tell `gofmt` to skip particular directories, so I temporarily just move them out of the source.